Mage/Module.cs

```diff
@@ -90,6 +90,11 @@ public class Module<T> : Singleton<T> where T : class, new() {
 	
 	public void setupUsercommands (Action<Exception> cb) {
 		logger.info ("Setting up usercommands");
+
+		if (commands == null) {
```

Comment:
Commands can only be one of 2 things, null or a list of strings. An empty list when iterated will iterate no elements, so it won't blow up. If the item isn't a list of strings it will blow up before compiling so don't have to implement checks.

As for why this is asynchronous, well it shouldn't be. The only reason why it is, is because these functions are unfortunately called from user land at the moment and async can be put into a nice async.series :(. The plan is to move these back into this classes land and call from there. In which case the async will be removed.